### ISSUE TITLE
Clarify preprocessor on/off behavior

### ISSUE THREAD

**Issue by rhine3** [kitzeslab/opensoundscape#325]
When turning actions on and off individually, they remain in the pipeline. You can't easily tell which options are on or off without checking the .bypass attribute of each action in the pipeline itself, e.g. `preprocessor.pipeline[5].bypass`.

But this behavior is not consistent with `CnnPreprocessor`'s behavior, where turning the augmentation on or off replaces the current pipeline with a different pipeline. This also overwrites any changes currently made to the pipeline.

It would be less confusing if the behaviors were consistent, e.g. turning an action off removes it from the pipeline, or using the `augmentation_off` behavior just turns those actions off. If the latter was used, it would be good to have a function for printing the pipeline (maybe connected to #324) which also prints the `bypass` state of each action in the pipeline

**Comment by sammlapp** [comment]
I agree there's undesireable inconsistency with CnnPreprocessors `.augmentation_on()` and `.augmentation_off()` - those two were added basically as helper functions to change between two pipelines. Perhaps they could be renamed to `select_pipeline(augmentation=True/False)`? The advantage is that you can still flip on/off switches and retain their positions, while also switching between augmentation and no augmentation.

**Comment by sammlapp** [comment]
Also, yes, in connection to #324 there should be a way to print the preprocessing pipeline as a table of 
```
action name | class | active/bypassed 
```

**Comment by sammlapp** [comment]
> I agree there's undesireable inconsistency with CnnPreprocessors `.augmentation_on()` and `.augmentation_off()` - those two were added basically as helper functions to change between two pipelines. Perhaps they could be renamed to `select_pipeline(augmentation=True/False)`? The advantage is that you can still flip on/off switches and retain their positions, while also switching between augmentation and no augmentation.

I'll leave this issue open until this aspect is resolved. #338 included a function to print the preprocessing pipeline (`preprocessor.pipeline_summary()`)
